### PR TITLE
Add support for Proto field predicates when populating with custom ra…

### PR DESCRIPTION
**Pull request by carborgar**
Fix for https://github.com/dvgaba/easy-random/issues/33 

Didn't work on the base repository as it involves exposing a property. Sorry it if was not the correct flow

### CODE REVIEW

**Comment by carborgar** [comment]
just let me know if there's something wrong

**Comment by dvgaba** [comment]
This isn't working, checking what needs to be done

`    @Test
    void shouldUseCustomProviderRegistryToFillFieldByType(){
        EasyRandomParameters parameters = new EasyRandomParameters()
            .randomize(ProtobufPredicates.ofProtobufType(
            Int32Value.class), () -> 42);
        EasyRandom easyRandom = new EasyRandom(parameters);
        Proto3Message protoInstance = easyRandom.nextObject(Proto3Message.class);
        assertThat(protoInstance.getInt32Field()).isEqualTo(42);
    }`

File: easy-random-protobuf/src/test/java/org/jeasy/random/protobuf/Protobuf3MessageGenerationTest.java [hunk 1]
```diff
@@ -210,4 +210,15 @@ void shouldUseCustomRandomizerForSubMessageWhenItsDefined() {
         assertThat(protoInstance.hasEmbeddedMessage()).isTrue();
         assertThat(protoInstance.getEmbeddedMessage()).isEqualTo(customEmbeddedMessage);
     }
+
+    @Test
+    void shouldUseCustomProviderRegistryToFillFields(){
```
Comment (dvgaba) [inline easy-random-protobuf/src/test/java/org/jeasy/random/protobuf/Protobuf3MessageGenerationTest.java:215]: IMO we need to add support for byType as well.

Comment (carborgar) [reply easy-random-protobuf/src/test/java/org/jeasy/random/protobuf/Protobuf3MessageGenerationTest.java:215]: will take a look, thanks for the feedback!

**Pull request closed without merging**